crates/flowctl/src/combine.rs

```diff
@@ -0,0 +1,307 @@
+use derive::combiner::Combiner;
+use doc::{ptr::Pointer, SchemaIndex, SchemaIndexBuilder, Validator};
+use futures::future::LocalBoxFuture;
+use models::tables::SchemaDoc;
+use protocol::flow::build_api;
+use std::io;
+use url::Url;
+
+#[derive(Debug, clap::Args)]
+pub struct CombineArgs {
+    #[clap(flatten)]
+    build_source: SchemaAndKeySource,
+    /// Maximum number of documents to add to the combiner before draining it. If 0, then there is no maximum
+    #[clap(long, default_value = "0")]
+    max_docs: u64,
+}
+
+/// How to get the schema and key
+#[derive(Debug, clap::Args)]
+pub struct SchemaAndKeySource {
+    /// Build directory
+    #[clap(long, default_value = ".")]
+    pub directory: String,
+    /// Path or URL to a JSON schema, which may include reduction annotations.
+    #[clap(long, conflicts_with_all(&["source", "collection"]), requires("key"))]
```

Comment:
Yeah, what I'd ideally like is to express these two sets of mutually exclusive arguments as an enum. I tried using somehting like this, but couldn't get it to work with clap:

```rust
pub enum SchemaAndKeySource {
  Direct {
    schema: String,
    key: Vec<String>,
  }
  Build {
    source: String,
    collection: String,
  }
}
```

I have a general sense that subcommands should be used for verbs, rather than for grouping arguments, which was why I went with the mutual exclusion flags. Ideally we'd be able to map those flags to different rust enums without requiring an extra subcommand, which feels like it should be doable.

I found [this thread](https://github.com/TeXitoi/structopt/issues/104) for structopt, which eventually led me to [this issue](https://github.com/clap-rs/clap/issues/2621), which seems like it'll eventually get us what we want. They have a proposed workaround there, too, which I can take a look at, though I'd prefer to do that in a follow on PR.